### ISSUE TITLE
Requesting the folder:PermissionSet attribute raises ErrorAccessDenied

### ISSUE THREAD

**Issue by driesken** [ecederstrand/exchangelib#610]
I was wondering if there are significant changes between the way Exchangelib 1.11.4 behaves and the current version (1.12.5). I've simply tried to upgrade using pip, but it appears that there is something wrong when authorizing in 1.12.5.

Some config, pretty basic:
```
from exchangelib import DELEGATE, IMPERSONATION, Account, Credentials, ServiceAccount, \
    EWSDateTime, EWSTimeZone, Configuration, NTLM, CalendarItem, Message, \
    Mailbox, Attendee, Q, ExtendedProperty, FileAttachment, ItemAttachment, \
    HTMLBody, Build, Version
from exchangelib.protocol import BaseProtocol, NoVerifyHTTPAdapter
BaseProtocol.HTTP_ADAPTER_CLS = NoVerifyHTTPAdapter

credentials = ServiceAccount(username='ai\\user', password='pass')
config = Configuration(server='mail.xxxxx.be', credentials=credentials, auth_type=NTLM)
tz_def = EWSTimeZone.timezone('Europe/Brussels')
account = Account(primary_smtp_address=xxxxx@xxxxx.be, config=config, autodiscover=False, default_timezone=tz_def, access_type=DELEGATE)
print(account.root.tree())
```

Is returning an "Access denied":
```
DEBUG:exchangelib.account:Added account: xxxxx@xxxxx.be
DEBUG:exchangelib.services:Getting folder Root (root)
DEBUG:exchangelib.services:Trying API version Exchange2016 for account xxxxx@xxxxx.be
DEBUG:exchangelib.protocol:Server mail.xxxxx.be: Waiting for session
DEBUG:exchangelib.protocol:Server mail.xxxxx.be: Got session 23594
DEBUG:exchangelib.util:Session 23594 thread 140058402891520: retry 0 timeout 120 POST'ing to https://mail.xxxxx.be/EWS/Exchange.asmx after 10s wait
/usr/local/lib/python3.6/site-packages/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
DEBUG:urllib3.connectionpool:https://mail.xxxxx.be:443 "POST /EWS/Exchange.asmx HTTP/1.1" 200 None
DEBUG:exchangelib.util:Retry: 0
Waited: 10
Timeout: 120
Session: 23594
Thread: 140058402891520
Auth type: <requests_ntlm.requests_ntlm.HttpNtlmAuth object at 0x7f61e39ddfd0>
URL: https://mail.xxxxx.be/EWS/Exchange.asmx
HTTP adapter: <exchangelib.protocol.NoVerifyHTTPAdapter object at 0x7f61e39ddef0>
Allow redirects: False
Streaming: False
Response time: 0.017957963049411774
Status code: 200
Request headers: {'User-Agent': 'python-requests/2.22.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'Keep-Alive', 'Content-Type': 'text/xml; charset=utf-8', 'Cookie': 'X-BackEndCookie=S-1-5-21-3222743990-3588243504-3938062828-46253=u56Lnp2ejJqBnMmbx8qdx5rSypvLztLLnsrJ0sfImcjSyMzJms7HycfHyZ7OgYHNz87G0s/H0s7Jq8/IxczIxc3GgZ6W0ZaRi5qNkZ6T0YqFmJqRi9GdmoHP; exchangecookie=79b6cb68921f405096eb97685de0ec78', 'Content-Length': '1169'}
Response headers: {'Cache-Control': 'private', 'Content-Type': 'text/xml; charset=utf-8', 'Content-Encoding': 'gzip', 'Vary': 'Accept-Encoding', 'Server': 'Microsoft-IIS/10.0', 'request-id': 'aa717744-a177-4c2d-94c7-7a1104a1bf7b', 'X-CalculatedBETarget': 'vmailboxsvr3p.xxxxx.be', 'X-DiagInfo': 'VMAILBOXSVR3P', 'X-BEServer': 'VMAILBOXSVR3P', 'X-AspNet-Version': '4.0.30319', 'Set-Cookie': 'exchangecookie=79b6cb68921f405096eb97685de0ec78; path=/, X-BackEndCookie=S-1-5-21-3222743990-3588243504-3938062828-46253=u56Lnp2ejJqBnMmbx8qdx5rSypvLztLLnsrJ0sfImcjSyMzJms7HycfHyZ7OgYHNz87G0s/H0s7Jq8/IxczIxcrGgZ6W0ZaRi5qNkZ6T0YqFmJqRi9GdmoHP; expires=Fri, 16-Aug-2019 07:37:59 GMT; path=/EWS; secure; HttpOnly', 'X-Powered-By': 'ASP.NET', 'X-FEServer': 'VMAILBOXSVR4P', 'Date': 'Wed, 17 Jul 2019 07:37:58 GMT', 'Transfer-Encoding': 'chunked'}
Request data: b'<?xml version=\'1.0\' encoding=\'utf-8\'?>\n<s:Envelope xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages" xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types"><s:Header><t:RequestServerVersion Version="Exchange2016"/><t:TimeZoneContext><t:TimeZoneDefinition Id="Romance Standard Time"/></t:TimeZoneContext></s:Header><s:Body><m:GetFolder><m:FolderShape><t:BaseShape>IdOnly</t:BaseShape><t:AdditionalProperties><t:FieldURI FieldURI="folder:ChildFolderCount"/><t:FieldURI FieldURI="folder:EffectiveRights"/><t:FieldURI FieldURI="folder:FolderClass"/><t:FieldURI FieldURI="folder:DisplayName"/><t:FieldURI FieldURI="folder:ParentFolderId"/><t:FieldURI FieldURI="folder:PermissionSet"/><t:FieldURI FieldURI="folder:TotalCount"/><t:FieldURI FieldURI="folder:UnreadCount"/></t:AdditionalProperties></m:FolderShape><m:FolderIds><t:DistinguishedFolderId Id="root"><t:Mailbox><t:EmailAddress>xxxxx@xxxxx.be</t:EmailAddress><t:RoutingType>SMTP</t:RoutingType><t:MailboxType>Mailbox</t:MailboxType></t:Mailbox></t:DistinguishedFolderId></m:FolderIds></m:GetFolder></s:Body></s:Envelope>'
Response data: b'<?xml version="1.0" encoding="utf-8"?><s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/"><s:Header><h:ServerVersionInfo MajorVersion="15" MinorVersion="1" MajorBuildNumber="845" MinorBuildNumber="34" Version="V2016_10_10" xmlns:h="http://schemas.microsoft.com/exchange/services/2006/types" xmlns="http://schemas.microsoft.com/exchange/services/2006/types" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"/></s:Header><s:Body xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><m:GetFolderResponse xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages" xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types"><m:ResponseMessages><m:GetFolderResponseMessage ResponseClass="Error"><m:MessageText>Access is denied. Check credentials and try again., Underlying MAPI stream threw exception</m:MessageText><m:ResponseCode>ErrorAccessDenied</m:ResponseCode><m:DescriptiveLinkKey>0</m:DescriptiveLinkKey><m:Folders/></m:GetFolderResponseMessage></m:ResponseMessages></m:GetFolderResponse></s:Body></s:Envelope>'

DEBUG:exchangelib.util:Session 23594 thread 140058402891520: Useful response from https://mail.xxxxx.be/EWS/Exchange.asmx
DEBUG:exchangelib.protocol:Server mail.xxxxx.be: Releasing session 23594
ERROR:root:Access is denied. Check credentials and try again., Underlying MAPI stream threw exception
```

**Comment by ecederstrand** [comment]
You are being authenticated correctly with NTLM because the server returns an XML response, but your GetFolder request for the root folder throws this error: `Access is denied. Check credentials and try again., Underlying MAPI stream threw exception`.

Google suggests this is a transient error on the server. You could try again later or report this to your Exchange admins.

This should also not work with older exchangelib versions, but you could try to download and try again. Maybe the server error and your upgrade just happened to coincide.

**Comment by driesken** [comment]
I've reversed to the older version of Exchangelib and it's working again. So I believe it must have something to do with Exchangelib?

**Comment by ecederstrand** [comment]
Hmm. Then your best bet is to compare the XML requests and responses between the two versions to see what the new version does differently. If you pretty-print the XML it's a bit easier to follow. See https://github.com/ecederstrand/exchangelib#troubleshooting

**Comment by driesken** [comment]
Okay, so I've compared the XML request between versions. At first sight, it looks like the new version is aditionally requesting:
`<t:FieldURI FieldURI="folder:PermissionSet"/>`

This is the older version (had to manually do some tabbing due to PrettyXML support missing in that release?):
```
Request data: <?xml version='1.0' encoding='utf-8'?>
<s:Envelope
	xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages"
	xmlns:s="http://schemas.xmlsoap.org/soap/envelope/"
	xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
	<s:Header>
		<t:RequestServerVersion Version="Exchange2007"/>
	</s:Header>
	<s:Body>
	<m:ResolveNames ReturnFullContactData="false">
		<m:UnresolvedEntry>ai\user</m:UnresolvedEntry>
	</m:ResolveNames></s:Body>
</s:Envelope>

```
Response data: 
```
<?xml version="1.0" encoding="utf-8"?>
<s:Envelope 
	xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
	<s:Header>
		<h:ServerVersionInfo MajorVersion="15" MinorVersion="1" MajorBuildNumber="845" MinorBuildNumber="34" 
			xmlns:h="http://schemas.microsoft.com/exchange/services/2006/types" 
			xmlns="http://schemas.microsoft.com/exchange/services/2006/types" 
			xmlns:xsd="http://www.w3.org/2001/XMLSchema" 
			xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"/>
	</s:Header>
	<s:Body xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
		<m:ResolveNamesResponse xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages" xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
		<m:ResponseMessages>
			<m:ResolveNamesResponseMessage ResponseClass="Error">
				<m:MessageText>No results were found.</m:MessageText>
				<m:ResponseCode>ErrorNameResolutionNoResults</m:ResponseCode>
				<m:DescriptiveLinkKey>0</m:DescriptiveLinkKey>
			</m:ResolveNamesResponseMessage>
		</m:ResponseMessages></m:ResolveNamesResponse>
	</s:Body>
</s:Envelope>
```

Request data: 
```
<?xml version='1.0' encoding='utf-8'?>
<s:Envelope 
	xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages" 
	xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" 
	xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
	<s:Header>
		<t:RequestServerVersion Version="Exchange2016"/>
		<t:TimeZoneContext>
			<t:TimeZoneDefinition Id="Romance Standard Time"/>
		</t:TimeZoneContext>
	</s:Header>
	<s:Body>
		<m:GetFolder>
			<m:FolderShape>
				<t:BaseShape>IdOnly</t:BaseShape>
				<t:AdditionalProperties>
					<t:FieldURI FieldURI="folder:ChildFolderCount"/>
					<t:FieldURI FieldURI="folder:EffectiveRights"/>
					<t:FieldURI FieldURI="folder:FolderClass"/>
					<t:FieldURI FieldURI="folder:DisplayName"/>
					<t:FieldURI FieldURI="folder:ParentFolderId"/>
					<t:FieldURI FieldURI="folder:TotalCount"/>
					<t:FieldURI FieldURI="folder:UnreadCount"/>
				</t:AdditionalProperties>
			</m:FolderShape>
			<m:FolderIds>
				<t:DistinguishedFolderId Id="root">
					<t:Mailbox>
						<t:EmailAddress>xxxxx@xxxxx.be</t:EmailAddress>
						<t:RoutingType>SMTP</t:RoutingType>
						<t:MailboxType>Mailbox</t:MailboxType>
					</t:Mailbox>
				</t:DistinguishedFolderId>
			</m:FolderIds>
		</m:GetFolder>
	</s:Body>
</s:Envelope>
```

Response data: 
```
<?xml version="1.0" encoding="utf-8"?>
	<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
		<s:Header>
			<h:ServerVersionInfo MajorVersion="15" MinorVersion="1" MajorBuildNumber="845" MinorBuildNumber="34" Version="V2016_10_10" 
				xmlns:h="http://schemas.microsoft.com/exchange/services/2006/types" 
				xmlns="http://schemas.microsoft.com/exchange/services/2006/types" 
				xmlns:xsd="http://www.w3.org/2001/XMLSchema" 
				xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"/>
		</s:Header>
		<s:Body 
			xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
			xmlns:xsd="http://www.w3.org/2001/XMLSchema">
			<m:GetFolderResponse xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages" xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
				<m:ResponseMessages><m:GetFolderResponseMessage ResponseClass="Success">
					<m:ResponseCode>NoError</m:ResponseCode>
					<m:Folders>
						<t:Folder>
							<t:FolderId Id="AAMkADlkYjZlNjM4LTJhYjEtNDZjMy1hZjk4LWYxY2RhOTY4MjJhMgAuAAAAAAD/14JL5sTwRKVwORBcMFFmAQDPfBWFqLlyTbUqssWKdVVmAAAABR9qAAA=" ChangeKey="AQAAABYAAADd/TnYqmuGQrA246n2jbz7AAAIv5po"/>
							<t:ParentFolderId Id="AAMkADlkYjZlNjM4LTJhYjEtNDZjMy1hZjk4LWYxY2RhOTY4MjJhMgAuAAAAAAD/14JL5sTwRKVwORBcMFFmAQDPfBWFqLlyTbUqssWKdVVmAAAABR9qAAA=" ChangeKey="AQAAAA=="/><t:DisplayName/>
							<t:TotalCount>13</t:TotalCount><t:ChildFolderCount>64</t:ChildFolderCount><t:EffectiveRights><t:CreateAssociated>false</t:CreateAssociated>
							<t:CreateContents>false</t:CreateContents>
							<t:CreateHierarchy>false</t:CreateHierarchy>
							<t:Delete>false</t:Delete>
							<t:Modify>false</t:Modify>
							<t:Read>false</t:Read>
							<t:ViewPrivateItems>false</t:ViewPrivateItems>
							</t:EffectiveRights>
							<t:UnreadCount>0</t:UnreadCount>
						</t:Folder>
					</m:Folders>
					</m:GetFolderResponseMessage>
				</m:ResponseMessages>
			</m:GetFolderResponse>
		</s:Body>
	</s:Envelope>
```


**And output of the lastest version:**

Request data: 
```
<?xml version='1.0' encoding='utf-8'?>
<s:Envelope
    xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages"
    xmlns:s="http://schemas.xmlsoap.org/soap/envelope/"
    xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
  <s:Header>
    <t:RequestServerVersion Version="Exchange2007"/>
  </s:Header>
  <s:Body>
    <m:ResolveNames ReturnFullContactData="false">
      <m:UnresolvedEntry>ai\user</m:UnresolvedEntry>
    </m:ResolveNames>
  </s:Body>
</s:Envelope>
```

Response data: 
```
<?xml version='1.0' encoding='utf-8'?>
<s:Envelope
    xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
  <s:Header>
    <h:ServerVersionInfo
    xmlns:h="http://schemas.microsoft.com/exchange/services/2006/types" xmlns="http://schemas.microsoft.com/exchange/services/2006/types"
    xmlns:xsd="http://www.w3.org/2001/XMLSchema"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" MajorVersion="15" MinorVersion="1" MajorBuildNumber="845" MinorBuildNumber="34"/>
  </s:Header>
  <s:Body
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:xsd="http://www.w3.org/2001/XMLSchema">
    <m:ResolveNamesResponse
    xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages"
    xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
      <m:ResponseMessages>
        <m:ResolveNamesResponseMessage ResponseClass="Error">
          <m:MessageText>No results were found.</m:MessageText>
          <m:ResponseCode>ErrorNameResolutionNoResults</m:ResponseCode>
          <m:DescriptiveLinkKey>0</m:DescriptiveLinkKey>
        </m:ResolveNamesResponseMessage>
      </m:ResponseMessages>
    </m:ResolveNamesResponse>
  </s:Body>
</s:Envelope>
```

Request data: 
```
<?xml version='1.0' encoding='utf-8'?>
<s:Envelope
    xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages"
    xmlns:s="http://schemas.xmlsoap.org/soap/envelope/"
    xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
  <s:Header>
    <t:RequestServerVersion Version="Exchange2016"/>
    <t:TimeZoneContext>
      <t:TimeZoneDefinition Id="Romance Standard Time"/>
    </t:TimeZoneContext>
  </s:Header>
  <s:Body>
    <m:GetFolder>
      <m:FolderShape>
        <t:BaseShape>IdOnly</t:BaseShape>
        <t:AdditionalProperties>
          <t:FieldURI FieldURI="folder:ChildFolderCount"/>
          <t:FieldURI FieldURI="folder:EffectiveRights"/>
          <t:FieldURI FieldURI="folder:FolderClass"/>
          <t:FieldURI FieldURI="folder:DisplayName"/>
          <t:FieldURI FieldURI="folder:ParentFolderId"/>
          <t:FieldURI FieldURI="folder:PermissionSet"/>
          <t:FieldURI FieldURI="folder:TotalCount"/>
          <t:FieldURI FieldURI="folder:UnreadCount"/>
        </t:AdditionalProperties>
      </m:FolderShape>
      <m:FolderIds>
        <t:DistinguishedFolderId Id="root">
          <t:Mailbox>
            <t:EmailAddress>xxxxx@xxxxx.be</t:EmailAddress>
            <t:RoutingType>SMTP</t:RoutingType>
            <t:MailboxType>Mailbox</t:MailboxType>
          </t:Mailbox>
        </t:DistinguishedFolderId>
      </m:FolderIds>
    </m:GetFolder>
  </s:Body>
</s:Envelope>

```
Response data: 
```
<?xml version='1.0' encoding='utf-8'?>
<s:Envelope
    xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
  <s:Header>
    <h:ServerVersionInfo
    xmlns:h="http://schemas.microsoft.com/exchange/services/2006/types" xmlns="http://schemas.microsoft.com/exchange/services/2006/types"
    xmlns:xsd="http://www.w3.org/2001/XMLSchema"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" MajorVersion="15" MinorVersion="1" MajorBuildNumber="845" MinorBuildNumber="34" Version="V2016_10_10"/>
  </s:Header>
  <s:Body
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:xsd="http://www.w3.org/2001/XMLSchema">
    <m:GetFolderResponse
    xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages"
    xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
      <m:ResponseMessages>
        <m:GetFolderResponseMessage ResponseClass="Error">
          <m:MessageText>Access is denied. Check credentials and try again., Underlying MAPI stream threw exception</m:MessageText>
          <m:ResponseCode>ErrorAccessDenied</m:ResponseCode>
          <m:DescriptiveLinkKey>0</m:DescriptiveLinkKey>
          <m:Folders/>
        </m:GetFolderResponseMessage>
      </m:ResponseMessages>
    </m:GetFolderResponse>
  </s:Body>
</s:Envelope>
```

**Comment by driesken** [comment]
I've removed the following line from /usr/local/lib/python3.6/site-packages/exchangelib/folders.py:
` PermissionSetField('permission_set', field_uri='folder:PermissionSet'),`

and it is not throwing the error anymore. Is this some permission issue on our Exchange servers?

**Comment by ecederstrand** [comment]
It's possible that the your server does not allow you to retrieve the `PermissionSet` value on the root folder, or possibly on any folder. I don't know if that's configurable and how to configure it, if so.

I'm not sure yet how we might handle this automatically, but for now a workaround that doesn't require you to patch source-code could be to monkey-patch the Folder class:

```python
from exchangelib import Folder
Folder.FIELDS = [f for f in Folder.FIELDS if f.name != 'permission_set']

# Your code here
```

**Comment by lululukas** [comment]
I've had the same problem as @driesken (albeit `GSSAPI`) and I can confirm that @ecederstrand 's workaround works.

**Comment by KoenR3** [comment]
I can also confirm that I had this problem and that the workaround works

**Comment by slith76** [comment]
I've had the same problem and the monkey-patch workaround works. But how can we solve the problem permanently? We are using Exchange 2016. Where can I configure the display of permission sets in Exchange?

**Comment by ecederstrand** [comment]
I think it would be useful to know if this is only a problem with the root folder, or with all folders. Can someone try this?

```python
from exchangelib import Folder
original_fields = Folder.FIELDS
Folder.FIELDS = [f for f in Folder.FIELDS if f.name != 'permission_set']
print(account.root)  # Should work. This is the workaround
Folder.FIELDS = original_fields
print(account.inbox)  # May work, unless permission_set is also not supported on other distinguished folders
print(account.root.tree())  # May work, if all folders except the root folder supports permission_set
```

**Comment by lululukas** [comment]
I’ve tested this and `print(account.inbox)` does indeed work with the original fields (returns `Inbox (Inbox)`).

`print(account.root.tree())` throws an error:

```python
AttributeError: 'Root' object has no attribute 'permission_set'
```

**Comment by ecederstrand** [comment]
Ah, that's another interesting bug. Can you post the full stack trace?

**Comment by lululukas** [comment]
Sorry about that, I forgot the `account.root.permission_set = None` line. With that, all of the print statements do work.

**Comment by ecederstrand** [comment]
Thanks for the extra test! I think I can provide a fix for this issue now.

**Comment by ecederstrand** [comment]
0a8f933 should fix this. I opted to only exclude `permission_set` for root folders on Exchange 2016 or later, as that is the only version I see reported here.

**Comment by agronholm** [comment]
I have a similar issue in #668 and I'm using exchangelib against Office 365 which reports this as version: <h:ServerVersionInfo MajorVersion="15" MinorVersion="20" MajorBuildNumber="2430" MinorBuildNumber="23" Version="V2018_01_08" ...>
Does that help?

**Comment by ecederstrand** [comment]
Hmm. That's weird. The field is deprecated from Exchange 2016 (see https://github.com/ecederstrand/exchangelib/blob/master/exchangelib/folders/roots.py#L34), and your version should resolve to Office 365 which is later than that.

**Comment by agronholm** [comment]
Ok, so if I understand correctly, the PermissionSet field should not actually be part of the query at all when dealing with O365, right? How can I debug this further?

**Comment by agronholm** [comment]
I don't get it. Now that I tried the latest version again, I get no error.

**Comment by ecederstrand** [comment]
Ah, sorry. I forgot that https://github.com/ecederstrand/exchangelib/commit/0a8f9337995f3c49dfe14aae62cdbeda73f48725 hadn't been released until I uploaded 2.1.0 earlier today.

**Comment by agronholm** [comment]
No worries. And thanks for the quick response!
